gcp-serverless/index.js

```diff
@@ -3,16 +3,28 @@ const config = require('./config.json')
 const { getColorCodes } = require('./getColorCodes')
 const callParticleApi = require('./callParticleApi')
 
-exports.callparticle = (request, response) => {
-  // @todo, does this var need to be escaped?
-  const deviceId = request.query.coreid
-  const particleToken = config.PARTICLE_TOKEN
-  const makeRequest = async () => {
-    const colorList = await getColorCodes(config.GITHUB_TOKEN)
-    await callParticleApi(particleToken, deviceId, colorList)
-    response.status(200).send(colorList)
+const makeRequest = async (coreid) => {
+  const colorList = await getColorCodes(config.GITHUB_TOKEN)
+
+  const configObject = {
+    'colorList': colorList,
+    // @todo, does this var need to be escaped?
+    'deviceId': coreid,
+    'particleToken': config.PARTICLE_TOKEN
+  }
+  return callParticleApi(configObject)
+}
+
+exports.makeRequest = makeRequest
+
+exports.callparticle = async (request, response) => {
+  const coreid = request.query.coreid
+  const returnVal = await makeRequest(coreid, response)
+  if (returnVal) {
+    response.status(200).send('success')
```

Comment:
Are these messages helpful return values? Previously the color codes were sent over http, yes?

Reply:
I don't have a strong opinion about what the best return text is, if any. The status codes are also debatable.

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Maybe return immediately with a `202`, regardless of whether the function is successful or not.

> 202 Accepted
> The request has been received but not yet acted upon. It is non-committal, meaning that there is no way in HTTP to later send an asynchronous response indicating the outcome of processing the request. It is intended for cases where another process or server handles the request, or for batch processing.

A `204` could also work for a success case.

> 204 No Content
> There is no content to send for this request, but the headers may be useful. The user-agent may update its cached headers for this resource with the new ones.

There should also be logic to handle a device id missing.

> 404 Not Found
> The server can not find requested resource. In the browser, this means the URL is not recognized. In an API, this can also mean that the endpoint is valid but the resource itself does not exist. Servers may also send this response instead of 403 to hide the existence of a resource from an unauthorized client. This response code is probably the most famous one due to its frequent occurence on the web.
>

Reply:
I opened #12. I'll follow up there.